Tester/Framework/Assert.php

```diff
@@ -60,11 +60,11 @@ public static function same($expected, $actual)
 	 * Checks assertion. Values must not be exactly the same.
 	 * @return void
 	 */
-	public static function notSame($expected, $actual)
+	public static function notSame($expected, $actual, $message = "")
 	{
 		self::$counter++;
 		if ($actual === $expected) {
-			self::fail('%1 should not be %2', $actual, $expected);
+			self::failMessage($message, '%1 should not be %2', $actual, $expected);
 		}
```

Comment:
Of course, name `join` is just suggestion. 

I am just trying to somehow solve problematic `failMessage`. We can rename it to failWithDescription (too ugly), we can add 4th argument to fail() (too weird), etc. 

Btw: to add color will require much more changes in code.